buffer/src/main/java/io/netty/buffer/UnpooledHeapByteBuf.java

```diff
@@ -517,6 +670,15 @@ protected void _setLong(int index, long value) {
         HeapByteBufUtil.setLong(array, index, value);
     }
 
+    @Override
+    public ByteBuf writeLongLE(long value) {
+        ensureAccessible();
+        int wIndex = writerIndex;
+        _setLongLE(wIndex, value);
```

Comment:
This and the other changes are not correct... we need to first ensure we can actually write 8 bytes as otherwise we might write into some memory that not belongs to us.

Reply:
I'm going to add this just in case - but I'll put it in draft: 
ATM since we check first vs `writerIndex` the JIT has to both verify (in batch, too) that array bounds are not surpassed AND that `writerIndex` is below such. 
In any case such comparison requires both information